### PR TITLE
clarify how to test configure/start/stop transitions

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

**Comment by g-arjones** [comment]
Documentation is good provided that the limitation is by design. I mean, I have a few different exceptions that may be thrown during configuration/start and right now there's no way to test whether the expected ones are being thrown for the given events.

**Comment by doudou** [comment]
Indeed. Actually getting it to work would not be a small piece of work (it's not only in the testing harness):

1. make sure that the orogen components "send" a state update **in all cases**
2. change Syskit to synchronize on the state update (it currently only synchronizes on the return of the configure call)
3. embed the state in the failed-to-start exception
4. add a `syskit_fail_to_configure` expectation

In addition, one would also need to add a `syskit_configure` action that is compatible with the testing harness (I don't think the current one is)

File: source/components/testing.html.md [hunk 1]
```diff
@@ -245,6 +245,58 @@ expect_execution { ACTIONS }
...
+
+~~~ ruby
+sample = expect_execution { task.start! }
+    .join_all_waiting_work(false)
+    .to { have_one_new_sample task.out_port }
+expect_execution { syskit_write task.in_port, data }
+    .join_all_waiting_work(false)
+    .to { have_one_new_sample task.out_port }
+expect_execution { syskit_write task.in_port, data }
+    .to { task.start_event }
```
Comment (g-arjones) [inline source/components/testing.html.md:297]:
```ruby
.to { emit task.start_event }
```

**Comment by g-arjones** [comment]
I think it's also worth mentioning that for the `have_no_new_sample` + `fail_to_start` case, one needs `validate_unexpected_errors(false)`

**Comment by doudou** [comment]
> I think it's also worth mentioning that for the have_no_new_sample + fail_to_start case, one needs validate_unexpected_errors(false)

This ... could be a bug. I don't see why one would need the `validate...` call

**Comment by g-arjones** [comment]
Could be... Can't investigate right now though, sorry.

File: source/components/testing.html.md (hunk 1 above)
Comment (doudou) [reply source/components/testing.html.md]: Done

**Pull request merged**